docs/DirectManipulation.md

```diff
@@ -0,0 +1,224 @@
+you will notice that it is a wrapper around `RCTUIManager.updateView` -
+this is the exact same function call that results from re-rendering -
+see [receiveComponent in
+ReactNativeBaseComponent.js](https://github.com/facebook/react-native/blob/master/Libraries/ReactNative/ReactNativeBaseComponent.js).
+
+## Composite components and setNativeProps
+
+Composite components are not backed by a native view, so you cannot call
+`setNativeProps` on them. Consider this example:
+
+```javascript
+var MyButton = React.createClass({
+  render() {
+    return (
+      <View>
+        <Text>{this.props.label}</Text>
+      </View>
+    )
+  },
+});
+
+var App = React.createClass({
+  render() {
+    return (
+      <TouchableOpacity>
+        <MyButton label="Press me!" />
+      </TouchableOpacity>
+    )
+  },
+});
+```
+
+If you run this you will immediately see this error: `Touchable child must either be native or forward setNativeProps to a native component`. This occurs because `MyButton` actually has no representation in the visible UI - there is no `UIView` for it, and you can't set the opacity of something that doesn't exist. You can think about it like this: if you define a component with `React.createClass` you would not expect to be able to set a style prop on it and have that work - you would need to pass the style prop down to a child, unless you are wrapping a native component. So the two options we have, as the error message so kindly suggests, are to make the composite component into a native component or to forward `setNativeProps` to child component with native backing (in the same way that we would pass style down to a child).
+
+#### Make a composite component native
+
+```javascript
+var NativeMethodsMixin = require('NativeMethodsMixin');
+var ReactNativeViewAttributes = require('ReactNativeViewAttributes');
+
+var MyButton = React.createClass({
+  mixins: [NativeMethodsMixin],
+
+  viewConfig: {
+    uiViewClassName: 'UIView',
+    validAttributes: ReactNativeViewAttributes.UIView,
+  },
+
+  render() {
+    return (
+      <View>
+        <Text>{this.props.label}</Text>
+      </View>
+    )
+  },
+});
+```
+
+`NativeMethodsMixin` adds `setNativeProps` and some other functions to
+`MyButton`, and the `viewConfig` property is used both by
+`NativeMethodsMixin` and internally by React.
+
+#### Forward setNativeProps to a child
+
+It's overkill to make every composite component that might need to
+support `setNativeProps` into a native-backed component and it should be
+avoided when possible because creating unnecessary `UIView`s can hurt
+memory usage and performance. So let's forward `setNativeProps` to a
+child component instead.
+
+```javascript
+var BUTTON_WRAPPER_REF = 'BUTTON_WRAPPER';
+
+var MyButton = React.createClass({
+  setNativeProps(nativeProps) {
+    this.refs[BUTTON_WRAPPER_REF].setNativeProps(nativeProps);
+  },
+
+  render() {
+    return (
+      <View ref={BUTTON_WRAPPER_REF}>
```

Comment:
@ide - sounds good, I'll reference [the React docs](https://facebook.github.io/react/docs/more-about-refs.html#the-ref-callback-attribute) in case people aren't familiar with that